pandas/core/series.py

```diff
@@ -378,103 +377,99 @@ def __init__(
                     data = SingleArrayManager.from_array(data, index)
             if copy:
                 data = data.copy()
-            if index is None:
```

Comment:
Why is this `if index is None:` check no longer needed in the `fastpath`?

Reply:
If you follow the code all the way down to the end of the method you can see that this actually is a noop; It's superflous code and can be removed. It was just difficult to see in the old code.

Reply:
Ah okay I see it now. Thanks